### ISSUE TITLE
Validation error messages

### ISSUE THREAD

**Issue by alyce** [magda-io/magda#2453]
## We need to design some accessible error messages for our Add Dataset fields that have form validation.

The two most relevant accessibility criteria are [3.3.1 Error Identification](https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-identified.html)
and [3.3.3 Error Suggestion](https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-suggestions.html) 

Some considerations, for usability and accessibility:

- inline error messages / validation warnings should appear as soon as input error is detected in a field (ideally either once the focus is placed on the next field or on immediate entry)
- if there are errors on a page, the user should not be able to proceed to the next screen, but be prompted to rectify any errors
- after pressing submit, and if there are errors, the focus should be moved to the first error message on the page, to help the user locate the error
- error messages should always be written clearly, without jargon, in positive framing, never blaming the user (please see [https://uxplanet.org/how-to-write-good-error-messages-858e4551cd4](https://uxplanet.org/how-to-write-good-error-messages-858e4551cd4))
- to meet accessibility criteria, we need to meet minimum colour contrast ratios for any colour used to show error
- colour alone should never be used to show an error; ideally a combination of a colour, symbol and text is used
- to meet accessibility criteria, inline error messages need to be appropriately coded, please review [https://hiddedevries.nl/en/blog/2017-04-04-how-to-make-inline-error-messages-accessible](https://hiddedevries.nl/en/blog/2017-04-04-how-to-make-inline-error-messages-accessible)
